src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -2761,6 +2761,102 @@ public static void DCS_TypeWithDelegate()
         Assert.Equal(value.IntProperty, actual.IntProperty);
     }
 
+    #region DesktopTest
+
+    /// <summary>
+    /// Roundtrips a Datacontract type  which contains Primitive types assigned to member of type object. 
+    /// Resolver is plugged in and resolves the primitive types. Verify resolver called during ser and deser
+    /// </summary>
+    [Fact]
+    public static void DCS_BasicRoundTripResolvePrimitiveTypes()
+    {
+        var dataContractSerializerSettings = new DataContractSerializerSettings()
+        {
+            DataContractResolver = new DesktopTestData.PrimitiveTypeResolver()
+            , IgnoreExtensionDataObject = false
+            , KnownTypes = null
+            , MaxItemsInObjectGraph = int.MaxValue
+            , PreserveObjectReferences = false
+        };
+
+        string baseline = ""; 
+        var value = new DesktopTestData.ObjectContainer(new DesktopTestData.PrimitiveContainer());
+
+        var actual = SerializeAndDeserialize(value, baseline, dataContractSerializerSettings, null, true);
+        // Throw Exception when verification failed
+        DesktopTestData.ComparisonHelper.CompareRecursively(value, actual);
+
+        var instance = new DesktopTestData.ObjectContainer(new DesktopTestData.PrimitiveContainer());
+
+        object result = DesktopTestData.DCRUtils.SingleRoundTripTest(
```

Comment:
What's the difference between `SingleRoundTripTest` and `SerializeAndDeserialize`? Are they verifying different scenarios?